docs/transpile/set-optimization.mdx

```diff
@@ -0,0 +1,154 @@
+
+- `optimization_level=0`: Trivial optimization, which maps the circuit to the backend with no explicit optimization.
+- `optimization_level=1-3`: Increasingly complex optimization, with heuristic algorithms that are used to find a layout and insert SWAP gates, with the goal of improving the overall performance of the circuit.  The number of iterations that these algorithms run increases with higher optimization levels. 
+
+Because finding the best layout is an NP-hard problem, it is the most time-consuming part of the transpilation process.  However, Qiskit uses stochastic algorithms that have been refactored into Rust, resulting in significant speedup.  Therefore, optimization levels 1-3 all use the same layout algorithms.  There are some slight differences in how the circuits are translated into basis gates, as described in the following table:
+
+
+<Table>
+  <thead>
+    <Tr>
+      <Th>Optimization Level</Th>
+      <Th>Description</Th>
+    </Tr>
+  </thead>
+  <tbody>
+    <Tr>
+      <Td>0</Td>
+      <Td>
+        No optimization: typically used for hardware characterization
+        - Basic translation
+        - Layout/Routing: `TrivialLayout`, where it selects the same physical qubit numbers as virtual and inserts SWAPs to make it work (using `StochasticSwap`)
+      </Td>
+    </Tr>
+    <Tr>
+      <Td>1</Td>
+      <Td>
+        Light optimization (default):
+        -   Layout/Routing: Layout is first attempted with `TrivialLayout`. If additional SWAPs are required, a layout with a minimum number of SWAPs is found by using `SabreSWAP`, then it uses `VF2LayoutPostLayout` to try to select the best qubits in the graph.
+        -   InverseCancellelation
+        -   1Q gate optimization
+      </Td>
+    </Tr>
+    <Tr>
+      <Td>2</Td>
+      <Td>
+        Medium optimization:
+          - Layout/Routing: Optimization level 1 (without trivial) + heuristic optimized with greater
+        search depth and trials of optimization function.  Because `TrivialLayout` is not used, there is no attempt to use the same physical and virtual qubit numbers. 
+        -   Commutative cancelation
+      </Td>
+    </Tr>
+    <Tr>
+      <Td>3</Td>
+      <Td>
+        High Optimization:
+        - Optimization level 2 + heuristic optimized on layout/routing further with greater effort/trials
+        - Resynthesis of two-qubit blocks using [Cartan's KAK Decomposition](https://arxiv.org/abs/quant-ph/0507171). 
+        - Unitarity-breaking passes: 
+          * `OptimizeSwapBeforeMeasure`: Moves the measurements around to avoid SWAPs
+          * `RemoveDiagonalGatesBeforeMeasure`: Removes gates before measurements that would not effect the measurements
+      </Td>
+    </Tr>
+  </tbody>
+</Table>
+
+Since CX is the noisiest gate, we can quantify the transpilation's "hardware efficiency" by counting the CX gates in the resulting circuit. We will compare the default transpilation levels given the same circuit. 
```

Comment:
Could you add a subheading straight after the table? It felt a bit jarring to go from reading the table and jumping straight into the next piece of info. Maybe add a subheading like:
```suggestion
## Optimization level in action

Since CX is the noisiest gate, we can quantify the transpilation's "hardware efficiency" by counting the CX gates in the resulting circuit. We will compare the default transpilation levels given the same circuit. 
```

or maybe "effect of optimization level on CX count" or whatever you think is best